services/github-service-api/src/main/java/org/kontinuity/catapult/service/github/api/GitHubRepository.java

```diff
@@ -12,7 +13,7 @@
     /**
      * @return the "git" protocol transport URL
      */
-    String getGitTransportUrl();
+    URI getGitTransportUrl();
```

Comment:
I suggest we use URI because URL rely on DNS resolution in some operations such as `equals`, which can be unnecessary expensive. See http://stackoverflow.com/questions/2348399/why-does-java-net-urls-hashcode-resolve-the-host-to-an-ip for example of discussion about it.